internal/services/apicenter/api_center_environment_resource.go

```diff
@@ -0,0 +1,304 @@
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			service, err := services.ParseServiceID(model.ServiceId)
+			if err != nil {
+				return err
+			}
+
+			// @favoretti: can't find any workspace creation buttons in the portal, assume everything defaults to "default" for now?
+			id := environments.NewEnvironmentID(subscriptionId, service.ResourceGroupName, service.ServiceName, "default", model.Identification)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing ApiCenter Environment %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			apiCenterEnvironmentProps := environments.EnvironmentProperties{
+				Kind:  environments.EnvironmentKind(model.Type),
+				Title: model.Name,
+			}
+
+			if model.Description != "" {
+				apiCenterEnvironmentProps.Description = pointer.To(model.Description)
+			}
+
+			apiCenterEnvironmentProps.Onboarding = &environments.Onboarding{}
+			if model.DevPortalUri != "" {
+				apiCenterEnvironmentProps.Onboarding.DeveloperPortalUri = pointer.To([]string{model.DevPortalUri})
+			}
+
+			if model.Instructions != "" {
+				apiCenterEnvironmentProps.Onboarding.Instructions = pointer.To(model.Instructions)
+			}
+
+			apiCenterEnvironmentProps.Server = &environments.EnvironmentServer{}
+
+			if model.ServerType != "" {
+				apiCenterEnvironmentProps.Server.Type = pointer.To(environments.EnvironmentServerType(model.ServerType))
+			}
+
+			if model.MgmtPortalUri != "" {
+				apiCenterEnvironmentProps.Server.ManagementPortalUri = pointer.To([]string{model.MgmtPortalUri})
+			}
+
+			apiCenterEnvironment := environments.Environment{
+				Name:       &model.Identification,
+				Properties: &apiCenterEnvironmentProps,
+				Type:       &model.Type,
+			}
+
+			if _, err = client.CreateOrUpdate(ctx, id, apiCenterEnvironment); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ApiCenterEnvironmentResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			id, err := environments.ParseEnvironmentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil || existing.Model == nil {
+				return fmt.Errorf("reading %s: %v", id, err)
+			}
+
+			model := existing.Model
```

Comment:
We're patching changes into properties so that should be nil checked as well, can you please update this to the suggestion below so it's consistent with other resources in the provider.
```suggestion
			if err != nil {
				return fmt.Errorf("retrieving %s: %v", id, err)
			}
			
			if resp.Model == nil {
				return fmt.Errorf("retrieving %s: `model` was nil", id)
			}
			if resp.Model.Properties == nil {
				return fmt.Errorf("retrieving %s: `properties` was nil", id)
			}

			model := existing.Model
```